### DISCUSSION TITLE
Export typescript schema

### DISCUSSION THREAD

**Discussion by Beiri22** [directus/directus#16867] [Feature Requests]
The directus SDK-manual suggests to use pass the database schema to the SDK constructor (https://docs.directus.io/reference/sdk.html#typescript)

To translate and maintain the schema by hand is a needless task, if you ask me. Directus should be able to export the current schema as ts-file automatically, maybe the "directus schema snapshot" command should get an option for typescript...

**Comment by paescuj** [comment]
See #5418

**Comment by rijkvanzanten** [comment]
This feature request has been accepted and is queued to be implemented! You can follow along with the progress here: https://github.com/directus/directus/issues/18121
